app/use_cases/update_shopping_cart.rb

```diff
@@ -0,0 +1,36 @@
+class UpdateShoppingCart
+  def run!(current_items:, target_items:)
+    item_types_to_keep = []
+    target_items.each do |i|
+      item_types_to_keep << i.item_type
+    end
+
+    current_items.each do |item|
+      unless item_types_to_keep.include? item.item_type
+        item.destroy!
+      end
+    end
+
+    target_items.each do |item_to_add|
```

Comment:
you can collect existing_items and new_items and call `RecalculatePrices.call(items:)` on existing ones and add new ones to current_items(you can use `push` method)